src/torchcodec/decoders/_core/CudaDevice.cpp

```diff
@@ -224,6 +224,9 @@ void convertAVFrameToDecodedOutputOnCuda(
 
   auto start = std::chrono::high_resolution_clock::now();
 
+  // TODO height and width info of output tensor comes from the metadata, which
+  // may not be accurate. How do we make sure we won't corrupt memory if the
+  // allocated tensor is too short/large?
```

Comment:
@ahmadsharif1 any idea on how to do this?

Reply:
You can look at rawOutput.frame's dimensions and use those, right?

Reply:
Are you suggesting to get height and width from the AVFrame instead of from the Metadata then? I don't mind but that's a change of logic. Do you know why it wasn't done like that before?

Reply:
discussed offline, I'll submit a follow-up to do that.